homeassistant/helpers/device_registry.py

```diff
@@ -43,29 +44,31 @@ def async_get_device(self, identifiers: str, connections: tuple):
         """Check if device is registered."""
         for device in self.devices:
             if any(iden in device.identifiers for iden in identifiers) or \
-                    any(conn in device.connection for conn in connections):
+                    any(conn in device.connections for conn in connections):
                 return device
         return None
 
     @callback
     def async_get_or_create(self, identifiers, manufacturer, model,
-                            connection, *, name=None, sw_version=None):
+                            connections, *, name=None, sw_version=None,
+                            config_entry=None):
```

Comment:
Let's make config entry mandatory (not default to None). Otherwise we don't know when to clean up devices etc. Now it's easy as on removal of a config entry we can delete things.

Reply:
Cleaning up devices on removal of a config entry is for another PR btw ;)

Reply:
I've been starting to look at removing devices in a separate PR 👍